balena/models/device.py

```diff
@@ -1032,8 +1032,38 @@ def set_to_release(self, uuid, commit_id):
 
         """
 
+        device = self.get(uuid)
+        query = {
+            'commit': commit_id,
+            'status': 'success',
+            'belongs_to__application': device['belongs_to__application']['__id']
+        }
+
+        release_id = self.release._Release__get_by_option(**query)[0]['id'] if commit_id else None
+
+        return self.set_to_release_by_id(uuid, release_id=release_id)
+
+    def set_to_release_by_id(self, uuid, release_id=None):
```

Comment:
Optionally, if a good amount of users are using this, it might be good idea to also check that the provided release_id belongs to a successful release.
PS: Probably by defining & reusing a private variant of set_to_release_by_id that doesn't have any check.

Reply:
@thgreasi I'm not sure about this one since in the support ticket, our users seem want to have a quick way to set the release without additional check. How about adding a method to verify if a release_id is a successful release and belongs to an application (optional) so user can do it manually if needed.

Reply:
I initially only checked for sbvr rules and there wasn't one, but I just found a hook in the open-balena-api that does the above check, so never mind. Let's keep as is, since the API will throw
> Release is not valid for this device